cli/src/opts/cast.rs

```diff
@@ -822,6 +822,27 @@ If an address is specified, then the ABI is fetched from Etherscan."#,
         about = "Runs a published transaction in a local environment and prints the trace."
     )]
     Run(RunArgs),
+    #[clap(name = "request")]
+    #[clap(visible_aliases = &["rq", "req"])]
+    #[clap(about = "Perform a raw JSON-RPC request")]
+    Request {
+        #[clap(short, long, env = "ETH_RPC_URL", value_name = "URL")]
+        rpc_url: Option<String>,
+        #[clap(value_name = "METHOD", help = "RPC method name")]
+        method: String,
+        #[clap(
+            value_name = "PARAMS",
+            help = "RPC parameters",
+            long_help = r#"RPC parameters
+
+Parameters are interpreted as strings. If you wish to pass a JSON value you can prepend the value with a ':'. To start a string with a ':' prepend with '::'. For example:
```

Comment:
I don't fully understand why this is necessary, it seems confusing imo.

why does `cast request doThing 123` not work?

Reply:
The problem is there's a little ambiguity if the parameters should be interpreted as a string or literal. Like if I pass 123, I don't know if that should be `123` or `"123"` and given that most params for JSON-RPC for ethereum are strings, I chose not to have to wrap each param in quotation marks and instead have a shorthand for interpreting things as not a string. But there might be a neater solution here. To be honest the double-colon  might be a little unnecessary for the rarer but ugly `':":param"'` edge-case...

Reply:
the other thing I left out of this PR is the possibility for named arguments in the future, but its also why I chose the ':' here because then we can implement named arguments like `arg1=hello arg2:=false` where the colon again references a literal argument instead of a string

Reply:
but in any case, I'll add the raw option

Reply:
I'd definitely be confused by the `:` tbh

`cast request eth_getBlockByNumber 0x123 false` should work 

since hex numbers are common can we do something like:

```rust
#[derive(serde::Deserialize, Debug)]
#[serde(untagged)]
enum Val {
    Value(serde_json::Value),
    Hash(TxHash),
    U256(U256),
    Number(U64),
}
```
?

Reply:
yeah, I think that makes sense. Maybe, we should just try to evaluate each parameter as a JSON first and if that doesn't work evaluate as string? I like your example but I think it misses the use case of something like `cast request eth_getBlockByNumber latest false` where we just want it to evaluate to `["latest", false]`. That would probably work for most use cases and I think its fairly easy to understand what its doing

Reply:
and if someone has to write `'"null"'` or `'"false"'` in some weird edge case, I guess they'll just have to deal with it